### ISSUE TITLE
Add includedContentTypes option to the compress middleware

### ISSUE THREAD

**Issue by ambroisemaupate** [traefik/traefik#9229]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you expect to see?

Compress middleware works with an `excludedContentTypes` list instead of an `included` (or whitelist) as you could find in Nginx or Apache configuration.

I believe there are more mime-type to exclude than to include and it can be cumbersome to add a very long list for all well-known mime-types to be already compressed:

- image/avif
- image/webp
- image/png
- image/jpeg
- font/woff2
- font/woff
- video/webm
- video/ogg
- video/mpeg
- video/mp4
- application/zip
- application/gzip
- ...

So is it possible to configure an `includedContentTypes` list ? Or not to break compatibility, include default values in  `excludedContentTypes` such as `application/grpc` (https://github.com/traefik/traefik/blob/master/pkg/middlewares/compress/compress.go#L33) ?

```yaml

# Add Gzip
- "traefik.http.middlewares.my_compress_middleware.compress=true"
- "traefik.http.middlewares.my_compress_middleware.compress.excludedcontenttypes=image/webp,image/png,image/jpeg,video/webm,font/woff2,video/ogg,video/mpeg,video/mp4,application/zip,application/gzip"
```

---

Related community posts:

- https://community.traefik.io/t/compression-included-content-types/6949

**Comment by jnoordsij** [comment]
This would be great! I think this makes more sense when configuring, as with include-based configuration it is way easier to find your most-used MIME types and pick those you want to compress. This also matches how other major server-tools like Apache and nginx implement this.

Possible non-breaking implementation: allow `includedContentTypes` as additional configuration key and compress all (non-excluded) content types when it is left empty (current default behavior).

**Comment by yongzhang** [comment]
Any plans on this? It would be much useful to have `includedContentTypes`.

**Comment by rjsocha** [comment]
I've just added a pull request that introduces the includedContentTypes option to the compress middleware. This new option complements the existing excludedContentTypes by allowing explicit listing of content types that should be compressed. I found this option to be incredibly useful in certain scenarios.

**Comment by motoki317** [comment]
`includedContentTypes` is available from v3.0 thanks to #10207 so this issue can probably be closed :)
